solutions/bronze/usaco-832.mdx

```diff
@@ -22,6 +22,184 @@ In this problem, there are three cases that we have to consider:
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int main() {
+	freopen("milkorder.in", "r", stdin);
+
+	int n, m, k;
+	cin >> n >> m >> k;
+
+	vector<int> hierarchy(m);
+	for (int i = 0; i < m; i++) { cin >> hierarchy[i]; }
+
+	vector<int> cows(n);
+	for (int i = 0; i < k; i++) {
+		int cow, position;
+		cin >> cow >> position;
+		cows[position - 1] = cow;
+	}
+
+	int ans;
+	if (find(cows.begin(), cows.end(), 1) != cows.end()) {
+		// Case 1
+		ans = find(cows.begin(), cows.end(), 1) - cows.begin() + 1;
+	} else {
+		if (find(hierarchy.begin(), hierarchy.end(), 1) != hierarchy.end()) {
+			// Case 2
+			int i = 0;
+			int pointer = 0;
+			while (pointer <= m - 1) {
+				if (find(cows.begin(), cows.end(), hierarchy[pointer]) !=
+				    cows.end()) {
+					// If the cow is in the hierarchy and has already been
+					// placed.
+					i = find(cows.begin(), cows.end(), hierarchy[pointer]) -
+					    cows.begin() + 1;
+					pointer++;
+				} else {
+					// Otherwise place the cow at the next earliest position.
+					if (cows[i] == 0) {
+						cows[i] = hierarchy[pointer];
+						pointer++;
+					}
+					i++;
+				}
+			}
+			for (int i = 0; i < n; i++) {
+				if (cows[i] == 1) {
+					ans = i + 1;
+					break;
+				}
+			}
+		} else {
+			// Case 3
+			int i = n - 1;
+			int pointer = m - 1;
+			while (pointer >= 0) {
+				if (find(cows.begin(), cows.end(), hierarchy[pointer]) !=
+				    cows.end()) {
+					// If the cow is in the hierarchy and has already been
+					// placed.
+					i = find(cows.begin(), cows.end(), hierarchy[pointer]) -
+					    cows.begin() - 1;
+					pointer--;
+				} else {
+					// Otherwise place the cow at the next latest position.
+					if (cows[i] == 0) {
+						cows[i] = hierarchy[pointer];
+						pointer--;
+					}
+					i--;
+				}
+			}
+			for (int i = 0; i < n; i++) {
+				if (cows[i] == 0) {
+					ans = i + 1;
+					break;
+				}
+			}
+		}
+	}
+
+	freopen("milkorder.out", "w", stdout);
```

Comment:
might wanna move this up js for cleanliness

Reply:
no i told him to do that